packages/openapi-specification/openapi.yaml

```diff
@@ -1,742 +1,177 @@
+        application/json:
+          schema:
+            $ref: '#/components/schemas/Record'
+
   schemas:
-    Order:
-      title: Pet Order
-      description: An order for a pets from the pet store
-      type: object
-      properties:
-        id:
-          type: integer
-          format: int64
-        petId:
-          type: integer
-          format: int64
-        quantity:
-          type: integer
-          format: int32
-        shipDate:
-          type: string
-          format: date-time
-        status:
-          type: string
-          description: Order Status
-          enum:
-            - placed
-            - approved
-            - delivered
-        complete:
-          type: boolean
-          default: false
-      xml:
-        name: Order
-    Category:
-      title: Pet category
-      description: A category for a pet
-      type: object
-      properties:
-        id:
-          type: integer
-          format: int64
-        name:
-          type: string
-          pattern: '^[a-zA-Z0-9]+[a-zA-Z0-9\.\-_]*[a-zA-Z0-9]+$'
-      xml:
-        name: Category
-    User:
-      title: a User
-      description: A User who is purchasing from the pet store
+    Record:
+      title: Website Record
+      description: A record of a website crawl
       type: object
+      required:
+        - url
+        - boundaryRegEx
       properties:
         id:
           type: integer
           format: int64
-        username:
-          type: string
-        firstName:
-          type: string
-        lastName:
-          type: string
-        email:
-          type: string
-        password:
+        url:
+          description: The URL at which the crawler should start.
           type: string
-        phone:
+          format: uri
+          pattern: '^https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&\/=]*)$'
+        boundaryRegEx:
+          description: The crawler will proceed crawling only the links that match this regular expression.
           type: string
-        userStatus:
-          type: integer
-          format: int32
-          description: User Status
-      xml:
-        name: User
-    Tag:
-      title: Pet Tag
-      description: A tag for a pet
-      type: object
-      properties:
-        id:
-          type: integer
-          format: int64
-        name:
+        periodicity:
+          description: How often should the site be crawled (minute, hour, day)
           type: string
-      xml:
-        name: Tag
-    Pet:
-      title: a Pet
-      description: A pet for sale in the pet store
-      type: object
-      required:
-        - name
-        - photoUrls
-      properties:
-        id:
-          type: integer
-          format: int64
-        category:
-          $ref: '#/components/schemas/Category'
-        name:
+          default: daily
+          enum:
+            - every minute
+            - hourly
+            - daily
```

Comment:
How would it be possible to set it for one hour and 30 minutes? Maybe we can convert it to periodicityInMinutes or periodicityInSeconds as int and therefore it would be simpler to calculate the next crawling time

Reply:
Or a cron string ( https://crontab.guru/#5_0_*_8_* )?

Reply:
I thought since in the assignment it says "Periodicity (minute, hour, day)" we were supposed to only take into account those options, but I must've misunderstood. It can be either cron string or int in seconds, I don't have any special feelings either way. What do you think would be better?